flax/core/frozen_dict.py

```diff
@@ -111,7 +111,11 @@ def __hash__(self):
       self._hash = h
     return self._hash
 
-  def copy(self, add_or_replace: Mapping[K, V]) -> 'FrozenDict[K, V]':
+  def copy(
+      self, add_or_replace: Optional[Mapping[K, V]] = None
+  ) -> 'FrozenDict[K, V]':
+    if add_or_replace is None:
+      add_or_replace = {}
```

Comment:
Another alternative could be
```suggestion
  def copy(
      self, add_or_replace: Mapping[K, V] = MappingProxyType({})
  ) -> 'FrozenDict[K, V]':
```
as `MappingProxyType` is immutable.